core/src/impl/KokkosExp_IterateTileGPU.hpp

```diff
@@ -87,65 +89,32 @@ struct DeviceIterateTile<2, PolicyType, Functor, MaxGridSize, Tag> {
 
   KOKKOS_IMPL_DEVICE_FUNCTION
   void exec_range() const {
-    // LL
-    if (PolicyType::inner_direction == Iterate::Left) {
-      // iterate over y blocks
-      for (index_type tile_id1 = static_cast<index_type>(blockIdx.y);
-           tile_id1 < m_policy.m_tile_end[1]; tile_id1 += gridDim.y) {
-        // compute index for dimension 1
-        const index_type offset_1 =
-            tile_id1 * m_policy.m_tile[1] +
-            static_cast<index_type>(threadIdx.y) +
-            static_cast<index_type>(m_policy.m_lower[1]);
-        // check index for dimension 1 is within range
-        if (offset_1 < m_policy.m_upper[1] &&
-            static_cast<index_type>(threadIdx.y) < m_policy.m_tile[1]) {
-          // iterate over x blocks
-          for (index_type tile_id0 = static_cast<index_type>(blockIdx.x);
-               tile_id0 < m_policy.m_tile_end[0]; tile_id0 += gridDim.x) {
-            // compute index for dimension 0
-            const index_type offset_0 =
-                tile_id0 * m_policy.m_tile[0] +
-                static_cast<index_type>(threadIdx.x) +
-                static_cast<index_type>(m_policy.m_lower[0]);
-            // check index for dimension 0 is within range
-            if (offset_0 < m_policy.m_upper[0] &&
-                static_cast<index_type>(threadIdx.x) < m_policy.m_tile[0]) {
-              // call kernel with computed indices
-              Impl::_tag_invoke<Tag>(m_func, offset_0, offset_1);
-            }
-          }
+    const index_type stride_0 = gridDim.x * blockDim.x;
+    const index_type stride_1 = gridDim.y * blockDim.y;
+
+    const index_type start_0 =
+        blockIdx.x * blockDim.x + threadIdx.x + m_policy.m_lower[0];
+    const index_type start_1 =
+        blockIdx.y * blockDim.y + threadIdx.y + m_policy.m_lower[1];
+
+    if constexpr (PolicyType::inner_direction == Iterate::Left) {
+      for (index_type offset_1 = start_1;
+           offset_1 < static_cast<index_type>(m_policy.m_upper[1]);
+           offset_1 += stride_1) {
+        for (index_type offset_0 = start_0;
+             offset_0 < static_cast<index_type>(m_policy.m_upper[0]);
+             offset_0 += stride_0) {
+          Impl::_tag_invoke<Tag>(m_func, offset_0, offset_1);
```

Comment:
```suggestion
      for (index_type idx_1 = start_1;
           idx_1 < static_cast<index_type>(m_policy.m_upper[1]);
           idx_1 += stride_1) {
        for (index_type idx_0 = start_0;
             idx_0 < static_cast<index_type>(m_policy.m_upper[0]);
             idx_0 += stride_0) {
          Impl::_tag_invoke<Tag>(m_func, idx_0, idx_1);
```

`offset_0` and `offset_1` seem to me just indices